server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/MailboxListenersLoaderImpl.java

```diff
@@ -0,0 +1,88 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.modules.mailbox;
+
+import java.util.Set;
+
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.apache.james.lifecycle.api.Configurable;
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.store.event.MailboxListenerRegistry;
+import org.apache.james.utils.ExtendedClassLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+
+public class MailboxListenersLoaderImpl implements Configurable, MailboxListenersLoader {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MailboxListenersLoaderImpl.class);
+
+    private final Injector injector;
+    private final MailboxListenerRegistry registry;
+    private final ExtendedClassLoader classLoader;
+    private final Set<MailboxListener> guiceDefinedListeners;
+
+    @Inject
+    public MailboxListenersLoaderImpl(Injector injector, MailboxListenerRegistry registry,
```

Comment:
Sorry if I was unclear about Injector use.
Here, you inject  `Injector`  for a single call `MailboxListener listener = injector.getInstance(clazz);`
In my opinion, it deserves a class `MailboxListenerFactory` with a single method `MailboxListener createInstance(clazz)` that would rely on the Injector.